src/main/java/com/rultor/agents/github/qtn/QnIfPull.java

```diff
@@ -77,16 +77,43 @@ public Req understand(final Comment.Smart comment,
         final URI home) throws IOException {
         final Issue.Smart issue = new Issue.Smart(comment.issue());
         final Req req;
-        if (issue.isPull()) {
-            req = this.origin.understand(comment, home);
-        } else {
+        if (QnIfPull.isNotPull(issue)) {
             new Answer(comment).post(
                 false,
                 QnIfPull.PHRASES.getString("QnIfPull.not-pull-request")
             );
-            req = Req.EMPTY;
+            req = Req.DONE;
```

Comment:
@dskalenko ok let me rephrase the question:

Please explain clear cut why in order to not have this code act on a closed PR we have to adjust the return signature for the `merge` command response outside of an actual PR ?
It seems to me that these things are fully unrelated.